### PR TITLE
Docs (adapted from Qulacs)

### PR DESCRIPTION
**Pull request by WrathfulSpatula**
@nathanshammah This is already fully adapted and working from the Qulacs repository. Really, users don't specifically need anymore information than this. (Does this look alright?)

### CODE REVIEW

**Comment by nathanshammah** [comment]
LGTM. Note that this and #13 are basically duplicate, modulo the correct explanations in simulator.rst and toml file in this one, and the readme edits I made to include the invisible markers to have the installation and support menu display, see [here](https://github.com/unitaryfund/pennylane-qrack/pull/13/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168f). Either way is fine for me. 

Note that automodapi in `code.rst`captures the init comment from `pennylane_qrack`, but not the qrack device module nor the docstrings or the QrackDevice class docstrings (maybe the docstrings are not formatted correctly, but not sure whether the `qrack_device.py` should show up in the API doc; changing toc tree to 2 or 3 did not help for me locally).

File: doc/installation.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.. include:: ../README.rst
```
Comment (cosenal) [inline doc/installation.rst:1]: @nathanshammah is this the idea we wanted to port from your PR #13?

File: doc/_static/puzzle_qulacs.png
Comment (cosenal) [inline doc/_static/puzzle_qulacs.png:1]: To be updated after #15

**Review by cosenal — COMMENTED** [review]
We should add a github workflow step to build the docs.

**Review by nathanshammah — CHANGES_REQUESTED** [review]
@WrathfulSpatula looks good, just remember to add the markers to the main `README.rst` file like done in #13. 

Note that the functions docstrings are not listed in the API doc, but the plugin is private. In any case, in another issue, we could at least format the QrackDevice class docstring, so that it shows up on the api page.

File: doc/installation.rst (hunk 1 above)
Comment (nathanshammah) [reply doc/installation.rst:1]: yes, the way it's in #13, which is copied from the Qulacs plugin, is that there are markers in the main [`README.rst`](https://github.com/unitaryfund/pennylane-qrack/blob/master/README.rst) file, that here need to be added, see [here](https://github.com/unitaryfund/pennylane-qrack/pull/13/files).

**Comment by nathanshammah** [comment]
@WrathfulSpatula any idea why tests are failing? Should they pass?

**Comment by WrathfulSpatula** [comment]
> Note that the functions docstrings are not listed in the API doc, but the plugin is private. In any case, in another issue, we could at least format the QrackDevice class docstring, so that it shows up on the api page.

@nathanshammah, actually, we can't. While we can change the single top-level comment, in order to display docs for `QrackSimulator`, the class would need to be exported as a public API component. However, it _shouldn't be_ exported publicly, because users never directly instantiate our classes or call any of our methods; this is already correct. We can alter that single string in the docs, or we could drop the API reference section, but users simply don't need to know anything about the API of `QrackSimulator`, by design of PennyLane.

File: doc/devices/simulator.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+be most appropriate for best performance. Qrack will also automatically use the "hybrid simulation algorithm"
+"layers" that tend to work best for general "BQP-complete" tasks like random circuit sampling.
+
+If you'd like to use a special-case algorithm, like quantum binary decision diagrams (QBDD), or if you want more
+direct control over choice of algorithm and hardware used, there are a series of boolean options users can toggle
+like so:
+
+.. code-block:: python
+
+    dev = qml.device('Qrack.simulator', wires=2, isBinaryDecisionTree=True)
```
[the remark this answers is not in the document]
Comment (WrathfulSpatula) [reply doc/devices/simulator.rst]: Fixed.

File: doc/installation.rst (hunk 1 above)
Comment (WrathfulSpatula) [reply doc/installation.rst:1]: Fixed.

**Review by nathanshammah — APPROVED** [review]
LGTM! 
Shall we open a separate issue to address @cosenal point on adding a "github workflow step to build the docs"? If it's easy to grab it from qulacs, let's do it, otherwise I'd move forward with merging this.

**Review by cosenal — APPROVED** [review]
LGTM

**Pull request merged**